### PR TITLE
test: remove renderModal util

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

- Removed `renderModal` util
- Cleaned up DOM between each test
- Simplified test config

Before:
image

After:
image

### Potential Breaking Changes

none

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/react-vapor/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/react-vapor/tsconfig.json [hunk 1]
```diff
@@ -2,13 +2,13 @@
...
         "baseUrl": ".",
         "paths": {
-            "@test-utils": ["src/TestUtils.tsx"],
+            "@test-utils": ["jest/utils.tsx"],
             "*": ["types/*", "*"]
         }
     },
-    "include": ["src"]
+    "include": ["src", "jest"],
+    "exclude": ["**/*.js"]
```
Comment (GermainBergeron) [inline packages/react-vapor/tsconfig.json:13]: This won't exclude the node_modules, dist and coverage folder 🤔

File: packages/react-vapor/src/components/modal/tests/ModalComposite.spec.tsx [hunk 2]
```diff
@@ -72,6 +72,24 @@ describe('ModalComposite', () => {
         expect(screen.getByRole('dialog', {name: 'second modal title'})).toBeVisible();
     });
 
+    it('finds the modal even if the modal title changes', async () => {
+        const ModalFixture = () => {
+            const [isLoading, setIsLoading] = React.useState(true);
+            React.useEffect(() => {
+                const timer = setTimeout(() => {
+                    setIsLoading(false);
+                }, 2000);
```
Comment (GermainBergeron) [inline packages/react-vapor/src/components/modal/tests/ModalComposite.spec.tsx:81]: Can you make the timeout shorter so the test run faster?

**Review by GermainBergeron — APPROVED** [review]

File: packages/react-vapor/tsconfig.json (hunk 1 above)
Comment (gdostie) [reply packages/react-vapor/tsconfig.json:13]: [`exclude` only exclude stuff that is included by `include`](https://www.typescriptlang.org/tsconfig#exclude). `node_modules`, `dist` and `coverage` folders aren't under either `src` or `jest`. So we are good here. You can confirm by running `tsc --showConfig` to see what files are included in compilation.

File: packages/react-vapor/src/components/modal/tests/ModalComposite.spec.tsx (hunk 2 above)
Comment (gdostie) [reply packages/react-vapor/src/components/modal/tests/ModalComposite.spec.tsx:81]: woops that wasn't supposed to be committed, my bad

**Review by aiheon — APPROVED** [review]

**Pull request merged**